src/middleware.ts

```diff
@@ -25,17 +26,27 @@ function determineProductSlug(req: NextRequest): string {
 	return '*'
 }
 
+function setHappyKitCookie(
+	cookie: Parameters<NextResponse['cookies']['set']>,
+	response: NextResponse
+): NextResponse {
+	response.cookies.set(...cookie)
+	return response
+}
+
 /**
  * Root-level middleware that will process all middleware-capable requests.
  * Currently used to support:
  * - Handling simple one-to-one redirects for .io routes
  * - Handling the opt in for cookie setting
  */
-export function middleware(req: NextRequest, ev: NextFetchEvent) {
+export async function middleware(req: NextRequest, ev: NextFetchEvent) {
+	let response: NextResponse
+	const edgeFlags = await getEdgeFlags({ request: req })
+	const { flags, cookie } = edgeFlags
```

Comment:
For now, would it be possible to limit this call to only the paths which are being tested?

Reply:
Is the concern adding additional latency to each request? Before we stop evaluating on each request, I'd like us to sync up with the HappyKit folks to make sure that's not going to break anything. I suspect it's fine, but I do know they chose this approach specifically to take advantage of upcoming features to reduce latency.

Reply:
@dstaley Yeah, exactly! Given that middleware runs on every single request, if we can avoid _any_ extra latency at this layer I think that is preferable.

Reply:
Updated https://github.com/hashicorp/dev-portal/pull/1190/commits/efbc3ce65166c793af8c6288d0e98a9589a96354